tower/src/util/clone_boxed.rs

```diff
@@ -0,0 +1,138 @@
+use super::ServiceExt;
+use futures_util::future::BoxFuture;
+use std::{
+    fmt,
+    task::{Context, Poll},
+};
+use tower_layer::{layer_fn, LayerFn};
+use tower_service::Service;
+
+/// A `Clone + Send` boxed `Service`.
```

Comment:
nit, TIOLI: should we maybe link to the docs for the `Clone` and `Service` traits?